inventory/environ.py

```diff
@@ -461,6 +462,38 @@ def getRandomString():
     char_set = string.ascii_uppercase + string.digits
     return ''.join(random.sample(char_set * 6, 6))
 
+def parseUrl(url, vars_scope):
```

Comment:
I attempted to use `from urlparse import urlparse` as well as `urllib3.utils` to parse this out, but I wasn't getting the correct hostname in certain cases. For instance:
```python
>> urlparse("lm1")
>> ParseResult(scheme='', netloc='', path='lm1', params='', query='', fragment='')
```
Would prefer not to write/maintain our own URL parser here, but IDK if there's anything else folks would recommend.

Reply:
yeah, `urlparse` and the likes only work for valid urls (complete with scheme and domain or at least `//` before the domain). So maybe we can use `urlparse` and then extract the items from the result only if `netscope` != '' , otherwise the entire url is the domain, and get the other items from `vars_scope`. We still need to write codes for this validation and whatnot, though.

Reply:
Yeah, I agree with @hendolim.  As long as it results in an overall decrease in lines of code we'll need to maintain.

Reply:
I followed your suggestions and cleaned this section up a bit while keeping the unit tests the same :P It's definitely an improvement I think!